examples/yaml_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"os"
+	"os/exec"
+)
+
+const (
+	timeoutSeconds = 600
+	sleepBetween   = 10
+)
+
+// cmd will run the command c with args and if input is provided, that will be piped
+// into the process as input
+func cmd(t *testing.T, c string, args []string, input string) (string, error) {
+	t.Helper()
+	binary, err := exec.LookPath(c)
+	if err != nil {
+		t.Fatalf("couldn't find %s, plz install", c)
+	}
+
+	t.Logf("Executing %s %v", binary, args)
+
+	cmd := exec.Command(binary, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
```

Comment:
Maybe we can use `CombinedOutput` instead?

Reply:
TIL! thanks :D

I'm not sure I do want to use combinedoutput in this case tho, cuz im using stderr if the command fails, and if not, im returning stdout. in the error case, dumping both seems fine, but in the success case, not including stderr seems like it makes sense to me. what do you think?